src/AutoRest.CSharp/Common/Output/Models/Types/ObjectTypeProperty.cs

```diff
@@ -193,33 +188,94 @@ internal string CreateExtraDescriptionWithManagedServiceIdentity()
-                {
-                    var objectTypes = valueType.Arguments.Where(arg => !arg.IsFrameworkType && arg.Implementation is ObjectType);
-                    if (objectTypes.Count() > 0)
-                    {
-                        var subDescription = objectTypes.Select(o => ((ObjectType)o.Implementation).CreateExtraDescriptionWithDiscriminator()).ToArray();
-                        updatedDescription = subDescription.Join("");
-                    }
-                }
+                return $"{propertyDescription}{binaryDataExtraDescription}";
             }
-            else if (valueType.Implementation is ObjectType objectType)
+
+            return $"{CreateDefaultPropertyDescription(Declaration.Name, IsReadOnly)}{binaryDataExtraDescription}";
+        }
+
+        private FormattableString CreateBinaryDataExtraDescription()
+        {
+            var type = Declaration.Type;
+            if (!type.IsFrameworkType)
+            {
+                return $"";
+            }
+
+            if (type.Equals(typeof(BinaryData)))
             {
-                updatedDescription = objectType.CreateExtraDescriptionWithDiscriminator();
+                return ConstructBinaryDataDescription("this property", InputModelProperty?.Type);
+            }
+
+            if (TypeFactory.IsList(type) && TypeFactory.GetElementType(type).Equals(typeof(BinaryData)))
+            {
+                return ConstructBinaryDataDescription("the element of this property", (InputModelProperty?.Type as InputListType)?.ElementType);
+            }
+
+            if (TypeFactory.IsDictionary(type) && TypeFactory.GetElementType(type).Equals(typeof(BinaryData)))
+            {
+                return ConstructBinaryDataDescription("the value of this property", (InputModelProperty?.Type as InputDictionaryType)?.ValueType);
+            }
+
+            return $"";
+        }
+
+        private FormattableString ConstructBinaryDataDescription(string typeSpecificDesc, InputType? inputType)
+        {
+            switch (inputType)
+            {
+                case InputPrimitiveType { Kind: InputTypeKind.Bytes or InputTypeKind.BytesBase64Url }:
+                    return $@"
+<para>
+To assign a byte[] to {typeSpecificDesc} use <see cref=""{typeof(BinaryData)}.FromBytes(byte[])""/>.
+The byte[] will be serialized to a Base64 encoded string.
+</para>
+<para>
+Examples:
+<list type=""bullet"">
+<item>
+<term>BinaryData.FromBytes(new byte[] {{ 1, 2, 3 }})</term>
+<description>Creates a payload of ""AQID"".</description>
+</item>
+</list>
+</para>";
+
+                default:
+                    return $@"
+<para>
+To assign an object to {typeSpecificDesc} use <see cref=""{typeof(BinaryData)}.FromObjectAsJson{{T}}(T, System.Text.Json.JsonSerializerOptions?)""/>.
+</para>
+<para>
+To assign an already formatted json string to this property use <see cref=""{typeof(BinaryData)}.FromString(string)""/>.
```

Comment:
hey I see you resolved this issue here: https://github.com/Azure/autorest.csharp/issues/3849

Could you add this issue into the issues your PR fixes? thank you